### PR TITLE
feat: rework agent split, workshop mode, focused commands (#322)

### LINKED ISSUE #322: Rework agent split, status:reworked label, workshop mode, focused commands, and agent language
## Problem

Five related gaps in the pipeline:

### 1. Rework conflation — one agent does two jobs

The Blacksmith handles both first-pass implementation and rework in a single agent file. After rework it sets `status:hammered` — the same label as first-pass completion. There's no way to distinguish "fresh implementation complete" from "rework complete." The Temperer has the same issue: it reviews both first-pass work and rework re-reviews with identical instructions, even though the Evaluator Philosophy section says to "verify specific feedback was addressed" on rework passes.

### 2. Agent resume drops the system prompt

`run_forge_agent` in `forge-lib.sh:828-834` treats `--resume` and `--agent` as mutually exclusive. On resume, only `--resume` is passed — the agent's system prompt is lost. Claude Code sessions preserve conversation history but NOT the agent system prompt. `--resume` and `--agent` can be combined (`claude --agent forge:X --resume Y`), and this is what enables the rework agent split.

### 3. No workshop mode

There's no way to do ad-hoc interactive work outside the queue. Sometimes you want to discuss an idea with the Blacksmith, have it file an issue for ledger/history tracking, implement it, and then have the Temperer review — without touching the pipeline state machine.

### 4. No focused commands

`forge hammer` and `forge temper` always pick the lowest open issue from the queue. There's no way to target a specific issue by number.

### 5. Agent language gaps

- **Blacksmith**: The user-perspective testing language ("navigate key pages affected by the change") is too shallow. The no-stubbing rule exists but the agent still has a rule to "file out-of-scope features" as follow-ups rather than expanding scope to fully implement.
- **Temperer**: Same shallow user-perspective language. Non-blockers on the first pass don't contribute to a REWORK verdict — the cheapest time to fix everything is before the rework loop starts.

### Part 1: Rework agent split

Create 4 new agents that handle the rework flow by resuming the original session with a different agent's system prompt (`--agent forge:X --resume <session-id>`):

| Agent | Mode | Input | Output | Role |
|-------|------|-------|--------|------|
| `Rework-Blacksmith` | interactive | `status:rework`, `status:needs-human` | `status:reworked` | Address Temperer feedback, human recovery |
| `auto-rework-blacksmith` | headless | `status:rework` | `status:reworked` or `status:needs-human` | Address feedback autonomously, escalate at 7+ cycles |
| `Rework-Temperer` | interactive | `status:reworked` | `status:tempered` or `status:rework` | Verify rework addressed feedback, check changed areas only |
| `auto-rework-temperer` | headless | `status:reworked` | `status:tempered` or `status:rework` | Same, headless |

**Simplify existing agents:**
- **Blacksmith / auto-Blacksmith**: Remove Section 2 "Human Recovery", Section 3 "Rework Detection", Step 11 "Address Rework Comments", rework ledger template, "Max rework cycles" rule. Step 13 always sets `status:hammered`.
- **Temperer / auto-Temperer**: Remove "Be proportional" rework-pass language from Evaluator Philosophy, rework history context from Step 4. Becomes purely a first-pass evaluator.

### Part 2: `status:reworked` label

New label in the state machine — distinguishes rework completion from first-pass:

```
First pass:   status:ready → Blacksmith → status:hammered → Temperer → status:tempered / status:rework
Rework:       status:rework → Rework-Blacksmith → status:reworked → Rework-Temperer → status:tempered / status:rework
```

Changes in `forge-lib.sh`:
- Add `"status:reworked|1d76db|Rework complete"` to `FORGE_REQUIRED_LABELS`
- Add `status:reworked` as `temperable` in `classify_lowest_open_issue`
- Update `run_stoke_loop`: route `status:rework` to `auto-rework-blacksmith`, route `status:reworked` to `auto-rework-temperer`
- Update `_blacksmith_prompt_for_status`: remove `status:rework` and `status:needs-human` (those route to different agents now)
- Fix `run_forge_agent`: pass `--agent` on resume too — stop treating `--resume` and `--agent` as mutually exclusive

All agent files (existing + new): add `--remove-label "status:reworked"` to every defensive label removal command.

### Part 3: Workshop mode

Interactive-only parallel flow that uses GitHub issues for ledger/history but stays off the pipeline state machine. No `status:*` labels — workshop issues get a `workshop` label plus descriptive labels (`type:*`, `scope:*`).

**New agents** (interactive only, no auto variants):
- `workshop-blacksmith.md` — Discuss with user, file a GitHub issue, then implement. Issue gets `workshop` label (no `ai-generated`, no `status:*`).
- `workshop-temperer.md` — Evaluate the workshop Blacksmith's work. PR/merge on approval. Rework is conversational (no label-driven loop).

**Session management**: Workshop sessions are tracked separately from pipeline sessions in `config.json` (under a `workshop` key, not mixed with `sessions`). Both `workshop-blacksmith` and `workshop-temperer` sessions track their issue number.

**CLI dispatch:**

| Command | State | Result |
|---------|-------|--------|
| `forge hammer new` | no workshop-blacksmith session | Start new Workshop-Blacksmith session |
| `forge hammer new` | session exists, issue not yet filed or still open | Resume session |
| `forge hammer new` | session exists, issue closed | Start new session |
| `forge temper new` | no workshop-blacksmith session | Exit: "Start a workshop with the Blacksmith first" |
| `forge temper new` | workshop-blacksmith session has no issue yet | Exit: "The Blacksmith hasn't filed an issue yet" |
| `forge temper new` | workshop-blacksmith has issue, no workshop-temperer session | Start new Workshop-Temperer session with that issue |
| `forge temper new` | workshop-blacksmith has issue, workshop-temperer session matches | Resume Workshop-Temperer session |
| `forge temper new` | workshop-blacksmith has issue, workshop-temperer has *different* issue | Start new Workshop-Temperer session (Blacksmith moved on) |
| `forge hammer new sessions` | any | Browse past workshop-blacksmith sessions |
| `forge temper new sessions` | any | Browse past workshop-temperer sessions |

**Workshop Temperer rework flow**: No `status:rework` labels. The Temperer tells you what needs fixing. You resume the Workshop-Blacksmith (`forge hammer new`) to address it, then re-run the Workshop-Temperer (`forge temper new`). On approval, the Temperer does PR/merge, which closes the issue. The next `forge hammer new` starts a fresh session.

### Part 4: Focused commands

```
forge hammer <N>     Target a specific pipeline issue by number
forge temper <N>     Target a specific pipeline issue by number
```

Auto variants unaffected — always pick from queue.

Changes:
- Add `classify_issue_by_number()` to `forge-lib.sh` — same classification logic as `classify_lowest_open_issue` but for a single issue. Case arms must stay in sync.
- Update `forge.sh` hammer/temper dispatch to parse optional numeric arg before falling through to queue-based dispatch
- For `status:rework`/`status:needs-human`, dispatch Rework-Blacksmith; for `status:reworked`, dispatch Rework-Temperer
- Update `show_usage()` and `show_command_help()`

### Part 5: Strengthen agent language

**Blacksmith + auto-Blacksmith — thorough user-perspective testing:**
Strengthen E2E test step: walk through affected pages thoroughly as a real user would — not a shallow spot check. Test the primary workflow end-to-end, then edge cases: empty states, error states, responsive behavior, navigation between pages, loading states. Verify changes haven't broken adjacent functionality. Consider the full scope of changes.

**Blacksmith + auto-Blacksmith — full implementation, no follow-ups:**
Remove the "File out-of-scope features only" rule. Replace with: expand scope to fully implement. If implementing the issue properly requires building supporting functionality not in the acceptance criteria, build it. No stubs, no follow-up issues, no deferral. Strengthen no-stubbing rule: no placeholder implementations, no empty function bodies, no hardcoded sample data standing in for real data, no disabled-by-default features.

**Temperer + auto-Temperer — thorough user-perspective evaluation:**
Strengthen browse step: walk through every affected page thoroughly. Use the feature end-to-end. Test happy path, then edge cases. Verify adjacent functionality wasn't broken.

**Temperer + auto-Temperer — non-blockers on first pass:**
On first pass (rework cycles = 0), non-blockers should contribute to a REWORK verdict. First pass is the cheapest time to address all findings. The must-fix vs non-blocker distinction is for rework re-reviews only (Rework-Temperer), not first-pass evaluation.

## Acceptance criteria

- [ ] `run_forge_agent` passes `--agent` on resume (not just `--resume`)
- [ ] `status:reworked` label defined in `FORGE_REQUIRED_LABELS`
- [ ] `classify_lowest_open_issue` classifies `status:reworked` as temperable
- [ ] `run_stoke_loop` routes `status:rework` → `auto-rework-blacksmith` and `status:reworked` → `auto-rework-temperer`
- [ ] 4 new rework agent files: `rework-blacksmith.md`, `auto-rework-blacksmith.md`, `rework-temperer.md`, `auto-rework-temperer.md`
- [ ] Blacksmith/auto-Blacksmith simplified (rework sections removed)
- [ ] Temperer/auto-Temperer simplified (rework proportionality removed)
- [ ] 2 new workshop agent files: `workshop-blacksmith.md`, `workshop-temperer.md` (interactive only)
- [ ] Workshop session management separate from pipeline sessions in config.json
- [ ] `forge hammer new` starts/resumes Workshop-Blacksmith (new session if issue closed)
- [ ] `forge temper new` starts/resumes Workshop-Temperer (validates blacksmith session and issue exist, starts new session if issue number diverges)
- [ ] `forge hammer new sessions` / `forge temper new sessions` browse past workshop sessions
- [ ] Workshop issues labeled `workshop` (no `ai-generated`, no `status:*`)
- [ ] `forge hammer <N>` targets a specific pipeline issue
- [ ] `forge temper <N>` targets a specific pipeline issue
- [ ] `classify_issue_by_number()` helper added with drift guard test
- [ ] All defensive `--remove-label` commands include `status:reworked` (all agent files)
- [ ] Strengthened user-perspective walkthrough language in Blacksmith and Temperer (all variants)
- [ ] "File out-of-scope features" rule replaced with "expand scope to fully implement" in Blacksmith variants
- [ ] First-pass Temperer treats non-blockers as must-fix
- [ ] Help text updated (`show_usage`, `show_command_help`)
- [ ] `CLAUDE.md` and `README.md` updated (labels, state machine, agent list, workshop docs)
- [ ] All existing tests pass + new tests for `status:reworked`, `classify_issue_by_number`, stoke loop routing, workshop session management

### PR DESCRIPTION
**Pull request by allan-mobley-jr**
## Summary

- **Rework agent split**: 4 new agents (Rework-Blacksmith, auto-Rework-Blacksmith, Rework-Temperer, auto-Rework-Temperer) handle rework by resuming the original session with `--agent` + `--resume`. Existing Blacksmith/Temperer simplified to first-pass only.
- **`status:reworked` label**: Distinguishes rework completion from first-pass. Rework-Temperer picks up `status:reworked`, regular Temperer picks up `status:hammered`.
- **Workshop mode**: `forge hammer new` / `forge temper new` for ad-hoc interactive work outside the pipeline queue. Workshop issues use a `workshop` label (no `status:*`). Session management with issue-number sync between Blacksmith and Temperer.
- **Focused commands**: `forge hammer <N>` / `forge temper <N>` target specific pipeline issues by number.
- **Agent resume fix**: `run_forge_agent` now always passes `--agent` (even on resume) so the agent system prompt is re-injected. This enables cross-agent session resumption.
- **Strengthened agent language**: Thorough user-perspective walkthroughs, "expand scope to fully implement" replaces "file out-of-scope features", first-pass Temperer treats all findings as blocking (no non-blocker category on first pass).

Closes #322

## Test plan

- [x] All 104 bats tests pass
- [ ] `forge doctor` in a test project confirms `status:reworked` and `workshop` labels created
- [ ] `forge hammer new` starts workshop, files issue, implements
- [ ] `forge temper new` picks up workshop issue
- [ ] `forge hammer <N>` targets a specific pipeline issue
- [ ] `forge temper <N>` targets a specific pipeline issue
- [ ] `forge hammer` (no args) still picks from queue
- [ ] Stoke loop routes `status:rework` to auto-rework-blacksmith
- [ ] Stoke loop routes `status:reworked` to auto-rework-temperer

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugin/agents/rework-blacksmith.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+**Important:** Gitignored writes (`.env.local`, `.claude/settings.local.json`, etc.) MUST be recorded here with `created (gitignored)` or `modified (gitignored)` — they're invisible to `git diff`, so this is the only audit trail.
+
+*Posted by the Forge Rework-Blacksmith.*"
+```
+
+### 11. Update Status
+
+```bash
+gh issue edit <N> --remove-label "status:ready" --remove-label "status:hammering" --remove-label "status:hammered" --remove-label "status:tempering" --remove-label "status:tempered" --remove-label "status:rework" --remove-label "status:reworked" --remove-label "status:needs-human" --add-label "status:reworked"
```
[the remark this answers is not in the document]
Comment (allan-mobley-jr) [reply plugin/agents/rework-blacksmith.md]: Fixed in b603fca. Removed the `--remove-label "status:reworked"` from both rework-blacksmith.md and auto-rework-blacksmith.md transition commands.

File: plugin/agents/workshop-blacksmith.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+)" \
+    --label "workshop" \
+    --label "type:<feature|bug|chore|refactor>" \
+    --label "scope:<ui|api|data|auth|infra|docs>"
+```
+
+**Do NOT add `ai-generated` or any `status:*` labels.** Workshop issues stay off the pipeline queue.
+
+Note the issue number — you'll use it for the branch, ledger, and all subsequent references.
+
```
[the remark this answers is not in the document]
Comment (allan-mobley-jr) [reply plugin/agents/workshop-blacksmith.md]: Fixed in b603fca. The CLI now auto-detects the workshop issue after the Workshop-Blacksmith session exits — scans for the most recent workshop-labeled issue and updates the session metadata via `update_workshop_session_issue`. No manual step needed.
